api/client/credentials.go

```diff
@@ -473,7 +475,10 @@ func (d *DynamicIdentityFileCreds) TLSConfig() (*tls.Config, error) {
 		},
 
 		// VerifyConnection is used instead of the static RootCAs field.
-		RootCAs: nil,
+		// However, there's some client code which relies on the static RootCAs
+		// field. So we set it to a copy of the current root CAs pool to support
+		// those - e.g ALPNDialerConfig.GetClusterCAs
+		RootCAs: d.tlsRootCAs.Clone(),
```

Comment:
Is there a practical way we could unit test this?

Reply:
I can add a rough unit test for this I believe - I'll do that shortly

Reply:
213597e082